lineapy/transformer/transform_code.py

```diff
@@ -22,9 +27,30 @@ def transform(
     if it exists.
     """
 
-    transformers = [
-        NodeTransformer(code, location, tracer),
-    ]
+    # create sourcecode object and register source code to db
+    src = SourceCode(id=get_new_id(), code=code, location=location)
+    tracer.db.write_source_code(src)
+
+    # defaults for executor that were set inside node transformer
+    # Set __file__ to the pathname of the file
+    if isinstance(location, Path):
+        tracer.executor.module_file = str(location)
+
+    # initialize the transformer IN ORDER of preference
+    transformers: List[BaseTransformer] = []
+    # python 3.7 handler
+    if sys.version_info < (3, 8):
+        transformers.append(Py37Transformer(src, tracer))
+    # python 3.8 handler
+    if sys.version_info < (3, 9):
+        transformers.append(Py38Transformer(src, tracer))
+
+    # main transformation handler
+    transformers.append(NodeTransformer(src, tracer))
+
+    # TODO control flow handler
```

Comment:
Should we not have the main transformation handler go last? Or do you plan on having the control flow handler work off the lineagraph?

Reply:
this isn't decided yet. I'm keeping the possibility open because it's only a matter of moving the exception for the default case that was previously inside generic_visitor of node_transformer. this is done in this PR. i'll leave the conditional transformer commented out for now.